temporal/api/cloud/nexus/v1/message.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package temporal.api.cloud.nexus.v1;
+
+option go_package = "go.temporal.io/api/cloud/nexus/v1;nexus";
+
+import "google/protobuf/timestamp.proto";
+
+message EndpointSpec {
+    // The name of the endpoint. Must be unique within an account.
+    // The name must match `[a-zA-Z_][a-zA-Z0-9_]*`.
+    // This field is mutable.
+    string name = 1;
+
+    // Indicates where the endpoint should forward received nexus requests to. 
+    EndpointTargetSpec target_spec = 2;
+
+    // The set of authorization policies for the endpoint. Each request's caller
+    // must match with at least one of the specs to be accepted by the endpoint.
+    // This field is mutable.
+    repeated EndpointAuthorizationSpec authorization_specs = 3;
+
+    // Optional description. Can be in markdown for rendering in the UI
+    string description = 4;
+}
+
+message EndpointTargetSpec {
+    // The type of target that this endpoint routes Nexus requests to.
+    // Possible values are: task_queue
+    string target_type = 1;
```

Comment:
note: this is essentially a poor person's version of `oneof`

Reply:
I'm fine with this but I'd like to understand why we can't use what's standard.

Reply:
will follow up with abhinav today just to have a solidified answer / reasoning here.

Reply:
Same, seems unnecessary that you can't use the presence of the next field to determine type

Reply:
Spoke with @anekkanti offline. as far as we can tell there are four options here:

1) use oneoff. I think we just decided not to use that because there were concerns about around how this manifests in other languages, and even the generated go code to read the field seems janky: https://protobuf.dev/reference/go/go-generated/#oneof. We were also worried that adding new types in the future would complicate versioning. Maybe both of these worries are unfounded. This is the first time we are adding something like this to the Cloud API, so worth laying the groundwork for this now. Curious if @cretz or @bergundy have thoughts on "oneoff" and cross-platform/language usage and any issues around that.

2) Don't use `oneoff`, but don't have the `target_type` field, just check that exactly one field is assigned a value. No reason we cannot do this. If I wanted to do a Get call for all endpoints of a specific target type, we would probably still need the string on that `GetNexusEndpoints` get request, but probably not a big deal. 

3) Use `target_type` as an `enum` (I think there was a previous discussion about not using enums, we can re-open that if need be).

4) Use `target_type` as string (as is now). It's a bit more explicit, but agree its unnecessary. We'd probably still include it as an optional filter on GetNexusEndpoints() [but it can be added later down the line since there is only one type for now anyway]

Leaning towards option 2 (just get rid of `target_type` field). Any concerns with that?

Reply:
I too lean towards option 2. Unrelated to this specifically, but I sure would like to re-open that never-use-proto-enum decision that was made long ago on this repo. I disagree with it.

Reply:
ok cool. removed it completely for now. we have only one type so we can figure out if we want to add target_type later for the search/get